adapters/playwire_ortb/playwire_ortb.go

```diff
@@ -0,0 +1,172 @@
+			bannerCopy := *request.Imp[i].Banner
+			if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+				format := bannerCopy.Format[0]
+				bannerCopy.W = &(format.W)
+				bannerCopy.H = &(format.H)
+			}
+			request.Imp[i].Banner = &bannerCopy
+			validImps = append(validImps, request.Imp[i])
+			trackingId = zone
+		} else if request.Imp[i].Video != nil {
+			err := validateVideoParams(request.Imp[i].Video)
+			if err != nil {
+				errs = append(errs, err)
+			} else {
+				validImps = append(validImps, request.Imp[i])
+				trackingId = zone
+			}
+		}
+	}
+
+	if len(validImps) == 0 {
+		return nil, errs
+	}
+
+	request.Imp = validImps
+
+	if request.Site != nil {
+		siteCopy := *request.Site
+		siteCopy.ID = trackingId
+		request.Site = &siteCopy
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     g.URI,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errs
+}
+
+// MakeBids unpacks the server's response into Bids.
+func (g *GumGumAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Bad user input: HTTP status %d", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: HTTP status %d", response.StatusCode),
+		}}
+	}
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: %d. ", err),
+		}}
+	}
+
+	var errs []error
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			mediaType := getMediaTypeForImpID(sb.Bid[i].ImpID, internalRequest.Imp)
+			if mediaType == openrtb_ext.BidTypeVideo {
+				price := strconv.FormatFloat(sb.Bid[i].Price, 'f', -1, 64)
+				sb.Bid[i].AdM = strings.Replace(sb.Bid[i].AdM, "${AUCTION_PRICE}", price, -1)
+			}
+
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &sb.Bid[i],
+				BidType: mediaType,
+			})
+		}
+	}
+
+	return bidResponse, errs
+}
+
+func preprocess(imp *openrtb.Imp) (string, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		err = &errortypes.BadInput{
+			Message: err.Error(),
+		}
+		return "", err
+	}
+
+	var gumgumExt openrtb_ext.ExtImpGumGum
```

Comment:
This piggybacks on the `ExtImpGumGum` type defined in `openrtb_ext/imp_gumgum.go`, correct?  I wonder if it would make sense to clone and use a `ExtImpPlaywireORTB` type in `openrtb_ext/imp_playwire_ortb.go` instead.  It's not as DRY, but it might be clearer since (if I'm understanding correctly) the adapter will be used for more than just GumGum OpenRTB.